extensions/big-data-cluster/src/bigDataCluster/dialog/addControllerDialog.ts

```diff
@@ -0,0 +1,494 @@
+	private rememberPwCheckBox: azdata.CheckBoxComponent;
+	private errorTextContainer: azdata.FlexContainer;
+
+	constructor(private model: AddControllerDialogModel) {
+	}
+
+	public showDialog(): void {
+		this.createDialog();
+		this.toggleSignInButton();
+		this.toggleErrorText();
+		azdata.window.openDialog(this.dialog);
+	}
+
+	private createDialog(): void {
+		this.dialog = azdata.window.createModelViewDialog(this.DialogTitle);
+		this.dialog.registerContent(async view => {
+			this.uiModelBuilder = view.modelBuilder;
+
+			let titledContainer = new TitledContainer(this.uiModelBuilder);
+			titledContainer.title = localize('bigDataClusters.addControllerDialog.signInToController', 'Sign In to Controller');
+			titledContainer.setTitleMargin(6, 0, 11, 0);
+			titledContainer.setPadding(15, 30, 0, 30);
+			titledContainer.contentTopLine = true;
+
+			let wrapperContainer: azdata.FlexContainer = this.uiModelBuilder.flexContainer()
+				.withLayout(<azdata.FlexLayout>{
+					flexFlow: 'column',
+					alignItems: 'stretch',
+					width: '100%'
+				})
+				.component();
+
+			let urlInputContainer: azdata.FlexContainer = this.uiModelBuilder.flexContainer()
+				.withLayout(<azdata.FlexLayout>{
+					flexFlow: 'row',
+					alignItems: 'center',
+					width: '100%'
+				})
+				.component();
+
+			let urlInputText = this.uiModelBuilder.text()
+				.withProperties<azdata.TextComponentProperties>({ value: 'URL' }).component();
+
+			this.urlInputBox = this.uiModelBuilder.inputBox()
+				.withProperties<azdata.InputBoxProperties>({ placeHolder: 'url', value: this.model.prefilledUrl }).component();
+			this.urlInputBox.onTextChanged(e => this.toggleSignInButton());
+
+			urlInputContainer.addItem(urlInputText, {
+				flex: '1, 1, 0%',
+				CSSStyles: {
+					'justify-content': 'flex-start',
+					'align-items': 'center',
+					'width': '80px'
+				}
+			});
+
+			urlInputContainer.addItem(this.urlInputBox, {
+				flex: '1, 1, 0%',
+				CSSStyles: {
+					'justify-content': 'flex-start',
+					'align-items': 'center',
+					'margin': '-2px 0px 0px 0px',
+					'width': '100%'
+				}
+			});
+
+			wrapperContainer.addItem(urlInputContainer, {
+				flex: '1, 1, 0%',
+				CSSStyles: {
+					'justify-content': 'flex-start',
+					'align-items': 'center',
+					'width': '100%'
+				}
+			});
+
+			let usernameInputContainer: azdata.FlexContainer = this.uiModelBuilder.flexContainer()
+				.withLayout(<azdata.FlexLayout>{
+					flexFlow: 'row',
+					alignItems: 'center',
+					width: '100%'
+				})
+				.component();
+
+			let usernameInputText = this.uiModelBuilder.text()
+				.withProperties<azdata.TextComponentProperties>({ value: 'Username' }).component();
+
+			this.usernameInputBox = this.uiModelBuilder.inputBox()
+				.withProperties<azdata.InputBoxProperties>({ placeHolder: 'username', value: this.model.prefilledUsername }).component();
+			this.usernameInputBox.onTextChanged(e => this.toggleSignInButton());
```

Comment:
localize (this and other instances in this PR)

Reply:
duplicated comment.